forgerock-auth/src/main/java/org/forgerock/android/auth/FROptions.kt

```diff
@@ -0,0 +1,134 @@
+package org.forgerock.android.auth
+
+import okhttp3.OkHttpClient
+
+data class FROptions(val server: Server, val oauth: OAuth, val service: Service, val urlPath: UrlPath, val sslPinning: SSLPinning) {
+    companion object {
+        // we cannot do a reference check here, because the reference could be different , but all the option value can be same
+        @JvmStatic
+        fun equals(old: FROptions?, new: FROptions?): Boolean {
+            return old?.oauth == new?.oauth
```

Comment:
Can we still do reference check + compare the value? so if the reference are same, we don't proceed to compare the value.
if (old !== new {
   //compare value 
} else {
   return true;
}

Reply:
sure, yeah we can do an extra check